src/components/Common/PlayButton/index.tsx

```diff
@@ -1,51 +1,104 @@
 import React, { ReactNode } from 'react';
 import ButtonWithDropdown from '../ButtonWithDropdown';
+import Trailer from '../../Trailer';
 
 interface PlayButtonProps {
   links: PlayButtonLink[];
 }
 
+interface PlayButtonState {
+  viewingUrl?: string;
+}
+
+export enum PlayButtonLinkTypes {
+  Web = 'web',
+  Trailer = 'trailer',
+}
+
 export interface PlayButtonLink {
+  type?: string;
   text: string;
   url: string;
   svg: ReactNode;
 }
 
-const PlayButton: React.FC<PlayButtonProps> = ({ links }) => {
-  if (!links || !links.length) {
-    return null;
+const PlayButton = class PlaybuttonClass extends React.Component<
```

Comment:
Why did you convert this to a class component?

Reply:
as my PR states, i have basic knowledge of react at best , but I believe it's because i've made it non function as im changing state within it. 

If this is possible via functional component still let me know and ill go read more docs

Reply:
It's very possible! You use `useState` to accomplish it. Take a look at our other components and look into React hooks. Class components are super bug-prone so we don't use them at all in Overseerr.

Reply:
Sounds good! I’ll go do that